examples/lb3-application/src/__tests__/acceptance/lb3app.acceptance.ts

```diff
@@ -0,0 +1,218 @@
+// Copyright IBM Corp. 2019. All Rights Reserved.
+// Node module: @loopback/example-lb3-application
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {OpenApiSpec} from '@loopback/rest';
+import {Client, expect} from '@loopback/testlab';
+import * as _ from 'lodash';
+import {CoffeeShopApplication} from '../../application';
+import {givenCoffeeShop, setupApplication} from './test-helper';
+
+const lb3App = require('../../../lb3app/server/server');
+
+describe('CoffeeShopApplication', () => {
+  let app: CoffeeShopApplication;
+  let client: Client;
+
+  before('setupApplication', async () => {
+    ({app, client} = await setupApplication());
+  });
+
+  after('closes application', async () => {
+    if (app) await app.stop();
+  });
+
+  context('basic REST calls for LoopBack 3 application', () => {
+    it('creates and finds a CoffeeShop', async () => {
+      const coffeeShop = givenCoffeeShop();
+      const response = await client
+        .post('/api/CoffeeShops')
+        .send(coffeeShop)
+        .expect(200);
+
+      expect(response.body).to.containDeep(
+        _.pick(coffeeShop, ['name', 'city']),
+      );
+
+      const result = await client.get(`/api/CoffeeShops/${response.body.id}`);
+      expect(result.body).to.containDeep(_.pick(coffeeShop, ['name', 'city']));
+    });
+
+    it("gets the CoffeeShop's status", async () => {
+      const response = await client.get('/api/CoffeeShops/status').expect(200);
+
+      expect(response.body.status).to.be.equalOneOf(
+        'We are open for business.',
+        'Sorry, we are closed. Open daily from 6am to 8pm.',
+      );
+    });
+
+    it('gets external route in application', async () => {
+      await client.get('/ping').expect(200, 'pong');
+    });
+  });
+
+  context('LoopBack 3 authentication', () => {
+    // tslint:disable-next-line:no-any
+    let User: any;
+
+    before(() => {
+      User = lb3App.models.User;
+    });
+
+    it('creates and logs in a User', async () => {
+      const user = await User.create({
+        email: 'created@email.com',
+        password: 'L00pBack!',
+      });
+
+      expect(user.email).to.eql('created@email.com');
+
+      await User.login({
+        email: 'created@email.com',
+        password: 'L00pBack!',
+      });
+    });
+
+    it('makes an authenticated request', async () => {
+      await User.create({
+        email: 'sample@email.com',
+        password: 'L00pBack!',
+      });
+
+      const token = await User.login({
+        email: 'sample@email.com',
+        password: 'L00pBack!',
+      });
+
+      const response = await client
+        .get(`/api/CoffeeShops/greet?access_token=${token.id}`)
+        .expect(200);
+
+      expect(response.body.undefined).to.eql('Hello from this Coffee Shop');
+    });
+
+    it.skip('rejects anonymous requests to protected endpoints', async () => {
```

Comment:
I'll update the skipped tests after https://github.com/strongloop/loopback-next/pull/2757 and https://github.com/strongloop/loopback-next/pull/2830 are landed.